File: 2328/2328.cs
public class Solution {
    public int NumberOfIncreasingPathsInAGrid(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(NumberOfIncreasingPathsInAGrid(root.left), NumberOfIncreasingPathsInAGrid(root.right));
    }
}

File: 2328/2328.py
class Solution:
    def number_of_increasing_paths_in_a_grid(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.number_of_increasing_paths_in_a_grid(root.left), self.number_of_increasing_paths_in_a_grid(root.right))